libs/community/langchain_community/embeddings/baichuan.py

```diff
@@ -77,17 +78,42 @@ def _embed(self, texts: List[str]) -> Optional[List[List[float]]]:
                 sorted_embeddings = sorted(embeddings, key=lambda e: e.get("index", 0))
                 # Return just the embeddings
                 return [result.get("embedding", []) for result in sorted_embeddings]
+            # Common Error 1: Invalid `api_key`
```

Comment:
why not using: 

```python
response.raise_for_status()
```

The warnings.warn seem like a pretty unusual pattern. You can improve the error message in the raised exception?

Reply:
get, i will modify it.